src/ApiCompat/Microsoft.DotNet.ApiCompatibility/Rules/AttributesMustMatch.cs

```diff
@@ -0,0 +1,263 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Collections.Specialized;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Runtime;
+using System.Text;
+using Microsoft.CodeAnalysis;
+using Microsoft.DotNet.ApiCompatibility.Abstractions;
+
+namespace Microsoft.DotNet.ApiCompatibility.Rules
+{
+    /// <summary>
+    /// This class implements a rule to check that the attributes between public members do not change.
+    /// </summary>
+    public class AttributesMustMatch : IRule
+    {
+        private readonly RuleSettings _settings;
+
+        private class AttributeGroup : IEqualityComparer<AttributeGroup>
+        {
+            public AttributeData? _repr;
+            public List<AttributeData?>? _attributes;
+            public List<bool>? _seen;
+            private RuleSettings _Settings;
```

Comment:
This should be readonly and lower cased: `_settings`.
```suggestion
            private readonly RuleSettings _settings;
```